test/mock_client.ts

```diff
@@ -34,7 +52,8 @@ class MockClient {
         };
     }
 
-    async search() {
+    async search(query: AnyObject) {
+        this.searchQuery = query;
```

Comment:
Since we are mocking the elasticsearch client, ee should validate the search request here (like elasticsearch would).